addons/mail/models/discuss/discuss_channel_member.py

```diff
@@ -187,6 +208,7 @@ def _discuss_channel_member_format(self, fields=None):
             fields = {
                 "channel": {},
                 "create_date": True,
+                "last_interest_dt": True,
```

Comment:
We should ideally not always return `last_interest_dt` on member.
The `last_interest_dt` of other members is of no concern, it should only be returned for the members of the current user.

Reply:
we actually did check the user in https://github.com/odoo/odoo/pull/155569/files#diff-cd763b90c80860ef917a5ec3a7ad2f2ec9dfd4a8067b5514d1d16b64c36d1696R241

Reply:
The reason we do the thing in the loop is making the batch operation possible, `_discuss_channel_member_format` used a lot with batch operation and we don't  want to slow the performance so I think it is better to check the user in the loop rather than do something like 
```py
for member in self:
            vals = {} // check the fields here which slows the performance
            formatted_member = member._discuss_channel_member_format(vals).get(member)
```

Reply:
But the parameter `fields` of this method is controlling what we return already, we shouldn't need to do an extra filtering inside it, the value should come from the caller.

We don't want to check if the member is the current user every time, we just "know" inside different flows if we send the full info or not, according to how we send the data (rpc return, specific user on bus).